clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -321,6 +321,99 @@ public RestrictedClassLoader() {
         }
     }
 
+    @Test
+    public void testIndirectConfigResolutionValidTestCases() {
+        Properties props = new Properties();
+
+        // Test Case: Valid Test Case for ConfigProviders as part of config.properties
+        props.put("config.providers", "file");
+        props.put("config.providers.file.class", "org.apache.kafka.common.config.provider.MockFileConfigProvider");
+        props.put("prefix.ssl.truststore.location.number", 5);
+        props.put("sasl.kerberos.service.name", "service name");
+        props.put("sasl.kerberos.key", "${file:/usr/kerberos:key}");
+        props.put("sasl.kerberos.password", "${file:/usr/kerberos:password}");
+        TestIndirectConfigResolution config = new TestIndirectConfigResolution(props);
+        assertEquals(config.originals().get("sasl.kerberos.key"), "testKey");
+        assertEquals(config.originals().get("sasl.kerberos.password"), "randomPassword");
+        assertEquals(config.originals().get("prefix.ssl.truststore.location.number"), 5);
+        assertEquals(config.originals().get("sasl.kerberos.service.name"), "service name");
+
+        // Test Case: Valid Test Case for ConfigProviders as a separate variable
+        Properties providers = new Properties();
+        providers.put("config.providers", "file");
+        providers.put("config.providers.file.class", "org.apache.kafka.common.config.provider.MockFileConfigProvider");
+        props = new Properties();
+        props.put("sasl.kerberos.key", "${file:/usr/kerberos:key}");
+        props.put("sasl.kerberos.password", "${file:/usr/kerberos:password}");
+        config = new TestIndirectConfigResolution(props, providers);
+        assertEquals(config.originals().get("sasl.kerberos.key"), "testKey");
+        assertEquals(config.originals().get("sasl.kerberos.password"), "randomPassword");
+
+    }
+
+    @Test
+    public void testIndirectConfigResolutionInValidTestCases() {
+        Properties props;
+        TestIndirectConfigResolution config;
+
+        // Test Case: Invalid class for Config Provider
+        props = new Properties();
+        props.put("config.providers", "file");
+        props.put("config.providers.file.class", "org.apache.kafka.common.config.provider.InvalidConfigProvider");
```

Comment:
Shouldn't this result in an exception?

Reply:
No if we are unable to resolve a value we will leave it unresolved rather then throwing an exception